pkg/envoy/lds/response.go

```diff
@@ -64,10 +64,10 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 
 	// --- INGRESS -------------------
 	// Apply an ingress filter chain if there are any ingress routes
-	if ingressRoutesPerHost, err := meshCatalog.GetIngressRoutesPerHost(proxyServiceName); err != nil {
-		log.Error().Err(err).Msgf("Error getting ingress routes per host for service %s", proxyServiceName)
+	if ingressInboundPolicies, err := meshCatalog.GetIngressPoliciesForService(proxyServiceName, svcAccount); err != nil {
+		log.Error().Err(err).Msgf("Error getting ingress inbound traffic policies for service %s", proxyServiceName)
 	} else {
-		thereAreIngressRoutes := len(ingressRoutesPerHost) > 0
+		thereAreIngressRoutes := len(ingressInboundPolicies) > 0
```

Comment:
Can we make this change when we start leveraging routes v2 or in a separate change instead? Just because this change pertains to adding support for ingress in routes v2 and not necessarily consuming it.

Reply:
@snehachhabria - Should this section be feature flagged to preserve the original code path?

Reply:
I have reverted this change and added a `TODO` to replace this with the new method as a part of the clean up